rudder-web/src/main/scala/com/normation/rudder/web/rest/directive/DirectiveAPIService2.scala

```diff
@@ -319,56 +317,52 @@ case class DirectiveAPIService2 (
     val actor = getActor(req)
     val directiveId = DirectiveId(id)
 
-    //Find existing directive
-    readDirective.getDirectiveWithContext(directiveId) match {
-      case Full((technique, activeTechnique, directive)) =>
-        restValues match {
-          case Full(restDirective) =>
-            // Update Technique
-            val updatedDirective = restDirective.updateDirective(directive)
-            // check Parameters value and version
-            val techniqueId = TechniqueId(technique.id.name, restDirective.techniqueVersion.getOrElse(technique.id.version))
-            val paramCheck = for {
-              paramEditor <- editorService.get(techniqueId, directiveId, updatedDirective.parameters)
-              check <- sequence (paramEditor.mapValueSeq.toSeq)( checkParameters(paramEditor))
-            } yield {
-              check
-            }
-            paramCheck match {
-              case Full(_) =>
-                val diff = ModifyToDirectiveDiff(technique.id.name,updatedDirective,technique.rootSection)
-                createChangeRequestAndAnswer(
-                    id
-                  , diff
-                  , technique
-                  , activeTechnique
-                  , updatedDirective
-                  , Some(directive)
-                  , actor
-                  , req
-                  , "Update"
-                )
-
-              case eb:EmptyBox =>
-                val fail = eb ?~ (s"Error with directive Parameters" )
-                val message = s"Could not update Directive ${directive.name} (id:${directiveId.value}): cause is: ${fail.msg}."
-                toJsonError(Some(directiveId.value), message)
-            }
-
-
-          case eb : EmptyBox =>
-            val fail = eb ?~ (s"Could extract values from request" )
-            val message = s"Could not modify Directive ${directiveId.value} cause is: ${fail.msg}."
-            toJsonError(Some(directiveId.value), message)
-        }
-
-      case eb:EmptyBox =>
-        val fail = eb ?~ (s"Could not find Directive ${directiveId.value}" )
+    ( for {
+      restDirective <- restValues ?~ (s"Could not extract values from request" )
+      // Find existing directive
+      (oldTechnique, activeTechnique, directive) <- readDirective.getDirectiveWithContext(directiveId) ?~ (s"Could not find Directive ${directiveId.value}" )
+
+      // Check if Technique version is chnaged (migration)
+      actualTechniqueId = TechniqueId(oldTechnique.id.name, restDirective.techniqueVersion.getOrElse(oldTechnique.id.version))
```

Comment:
you mean updated in this case (even if the 'oldTechnique.id.name' got me there for a minute)